app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -711,7 +711,7 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
+            if (typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount) {
```

Comment:
Thanks @OZZlE , you are right about unnecessary strict comparison.

@zetlen, 'result' is undefined before all of the swatches are selected. It was the issue as with oldPrice and also with tierPrices. (how to reproduce is described in 8923 pull request)